z/ctrmv stack allocation failed on AMD bulldozer and barcelona target

http://build.openblas.net/builders/bulldozer-linux/builds/6/steps/shell_1/logs/stdio

```
cblat2: ztrmv.c:265: ctrmv_: Assertion `stack_check == 0x7fc01234' failed.
/bin/sh: line 1: 18952 Aborted                 (core dumped) OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./cblat2 < ./cblat2.dat
make: *** [level2] Error 134
```

http://build.openblas.net/builders/barcelona-linux/builds/4/steps/shell_1/logs/stdio
```
OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./zblat2 < ./zblat2.dat
zblat2: ztrmv.c:265: ztrmv_: Assertion `stack_check == 0x7fc01234' failed.
/bin/sh: line 1: 18947 Aborted                 (core dumped) OPENBLAS_NUM_THREADS=1 OMP_NUM_THREADS=1 ./zblat2 < ./zblat2.dat
make: *** [level2] Error 134
```

One walk around fix is replacing the old assembly by C kernel.

@jeromerobert

Reproduce:

```
git clone https://github.com/xianyi/OpenBLAS-CI
cd OpenBLAS-CI
./run_test_helper.sh quick_test_linux.sh "TARGET=BULLDOZER"
```

--- a/driver/level2/ztrmv_L.c
+++ b/driver/level2/ztrmv_L.c
@@ -56,7 +56,7 @@ int CNAME(BLASLONG m, FLOAT *a, BLASLONG lda, FLOAT *b, BLASLONG incb, FLOAT *bu
 
   if (incb != 1) {
     B = buffer;
-    gemvbuffer = (FLOAT *)(((BLASLONG)buffer + m * sizeof(FLOAT) * 2 + 4095) & ~4095);
+    gemvbuffer = (FLOAT *)(((BLASLONG)buffer + m * sizeof(FLOAT) * 2 + 15) & ~15);
     COPY_K(m, b, incb, buffer, 1);
   }
 

